rootfs/api/models/app.py

```diff
@@ -129,7 +132,65 @@ def delete(self, *args, **kwargs):
         self._clean_app_logs()
         return super(App, self).delete(*args, **kwargs)
 
-    def restart(self, **kwargs):
+    def restart(self, **kwargs):  # noqa
+        """
+        Restart found pods by deleting them (RC will recreate).
+        Wait until they are all drained away and RC has gotten to a good state
+        """
+        try:
+            # Fetch the initial set of pods to work from
+            pods = self.list_pods(**kwargs)
+            desired = len(pods)
+        except KubeException:
+            # Nothing was found
+            return []
+
+        try:
+            for pod in pods:
```

Comment:
whichever works. I'd just rather not block requests for longer than we require (e.g. if the pods take the maximum amount of time to shut down, `ps:restart` will be significantly slower than the old `ps:restart`)